public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/SimpleCondition.tsx

```diff
@@ -71,24 +73,37 @@ export const SimpleConditionEditor = ({
     updateThresholdFunction(value.value ?? EvalFunction.IsAbove, expressionQueriesList, dispatch);
   };
 
-  const onEvaluateValueChange = (event: FormEvent<HTMLInputElement>, index?: number) => {
-    if (isRange) {
+  const onEvaluateValueChange = (event: FormEvent<HTMLInputElement>, index = 0) => {
+    const value = event.currentTarget.value;
+
+    // Allow empty value
+    if (value === '') {
       const newParams = produce(simpleCondition.evaluator.params, (draft) => {
-        draft[index ?? 0] = parseFloat(event.currentTarget.value);
+        draft[index ?? 0] = undefined;
       });
-      // update the condition kept in the parent
       onChange({ ...simpleCondition, evaluator: { ...simpleCondition.evaluator, params: newParams } });
-      // update the reducer state where we store the queries
-      updateThresholdValue(parseFloat(event.currentTarget.value), index ?? 0, expressionQueriesList, dispatch);
-    } else {
-      // update the condition kept in the parent
-      onChange({
-        ...simpleCondition,
-        evaluator: { ...simpleCondition.evaluator, params: [parseFloat(event.currentTarget.value)] },
+      return;
+    }
+    // Allow '-' for negative numbers
+    if (value === '-') {
+      const newParams = produce(simpleCondition.evaluator.params, (draft) => {
+        draft[index ?? 0] = '-';
```

Comment:
This is quite confusing. We set `draft[index] = '-'`, so where we set the actual value?

Reply:
we set the actual value (in this case `-`, in `simpleCondition.evaluator.params[index]`